windows/src/main/csharp/ch/cyberduck/ui/controller/TransferController.cs

```diff
@@ -16,731 +16,330 @@
             {
-                View.TranscriptVisible = _preferences.getBoolean("queue.transcript.open");
-                View.TranscriptHeight = _preferences.getInteger("queue.transcript.size.height");
-
-                View.ToggleTranscriptEvent += View_ToggleTranscriptEvent;
-                View.TranscriptHeightChangedEvent += View_TranscriptHeightChangedEvent;
-            };
-            View.QueueSize = _preferences.getInteger("queue.connections.limit");
-            View.BandwidthEnabled = false;
-
-            View.ResumeEvent += View_ResumeEvent;
-            View.ReloadEvent += View_ReloadEvent;
-            View.StopEvent += View_StopEvent;
-            View.RemoveEvent += View_RemoveEvent;
-            View.CleanEvent += View_CleanEvent;
-            View.OpenEvent += View_OpenEvent;
-            View.ShowEvent += View_ShowEvent;
-            View.TrashEvent += View_TrashEvent;
-            View.SelectionChangedEvent += View_SelectionChangedEvent;
-            View.BandwidthChangedEvent += View_BandwidthChangedEvent;
-            View.ConnectionsChangedEvent += ViewConnectionsChangedEvent;
-
-            View.ValidateResumeEvent += View_ValidateResumeEvent;
-            View.ValidateReloadEvent += View_ValidateReloadEvent;
-            View.ValidateStopEvent += View_ValidateStopEvent;
-            View.ValidateRemoveEvent += View_ValidateRemoveEvent;
-            View.ValidateCleanEvent += View_ValidateCleanEvent;
-            View.ValidateOpenEvent += View_ValidateOpenEvent;
-            View.ValidateShowEvent += View_ValidateShowEvent;
+                window.Dispatcher.Invoke(window.Close);
+            }
         }
 
-        private void View_TrashEvent()
+        public override void invoke(MainAction action, bool wait)
         {
-            foreach (IProgressView progressView in View.SelectedTransfers)
+            if (!action.isValid())
             {
-                Transfer transfer = GetTransferFromView(progressView);
-                if (!transfer.isRunning())
-                {
-                    for (int i = 0; i < transfer.getRoots().size(); i++)
-                    {
-                        TransferItem item = (TransferItem)transfer.getRoots().get(i);
-                        try
-                        {
-                            LocalTrashFactory.get().trash(item.local);
-                        }
-                        catch (Exception exception)
-                        {
-                            Log.warn(String.Format("Failure trashing file {0} {1}", item.local, exception.Message));
-                        }
-                    }
-                }
+                return;
             }
-        }
 
-        private void View_TranscriptHeightChangedEvent()
-        {
-            _preferences.setProperty("queue.transcript.size.height", View.TranscriptHeight);
+            if (wait)
+            {
+                MainController.Application.SynchronizationContext.Send(
```

Comment:
In order to not get bitten by not-supported UI display scaling when opening prompts from the Transfer-window thread which has DPI display scaling enabled, schedule everything from background actions to WinForms main thread.